instrumentation/github.com/gorilla/mux/otelmux/mux.go

```diff
@@ -86,7 +87,19 @@ type traceware struct {
 }
 
 // defaultSpanNameFunc just reuses the route name as the span name.
-func defaultSpanNameFunc(routeName string, _ *http.Request) string { return routeName }
+func defaultSpanNameFunc(routeName string, r *http.Request) string {
+	method := r.Method
+	if !slices.Contains([]string{
+		http.MethodGet, http.MethodHead,
+		http.MethodPost, http.MethodPut,
+		http.MethodPatch, http.MethodDelete,
+		http.MethodConnect, http.MethodOptions,
+		http.MethodTrace,
+	}, method) {
```

Comment:
Why use a slice here instead of a `map`?


```suggestion
var validMethods = map[string]struct{}{
	http.MethodGet: {},
	http.MethodHead: {},
	http.MethodPost: {},
	http.MethodPut: {},
	http.MethodPatch: {},
	http.MethodDelete: {},
	http.MethodConnect: {},
	http.MethodOptions: {},
	http.MethodTrace: {},
}

// defaultSpanNameFunc just reuses the route name as the span name.
func defaultSpanNameFunc(routeName string, r *http.Request) string {
	method := r.Method
	if _, ok := validMethods[method]; !ok {
```

The `O(1)` seems preferable to the `O(N)`